app/controllers/drops_controller.rb

```diff
@@ -73,11 +73,11 @@ def set_drop
     end
 
     def drop_params
-      params.require(:drop).permit(:sc_track, :title)
+      params.require(:drop).permit(:sc_track, :title, :latitude, :longitude)
     end
 
     def place_params
-      params.require(:drop).require(:place).permit(:name, :longitude, :latitude, :location)
+      params.require(:drop).permit(place: [:name, :longitude, :latitude, :location]).fetch(:place, {})
```

Comment:
Yes, leaving it at `require` would return an error if the `params` hash didn't include `place`-data. `permit` makes it optional, so to speak.

The `Hash#fetch` directive is safer (or at least: more predictable) way to access a hash by key when you don't know if said key actually exists in the hash. The second argument is the "default" value that should be returned when the key is not present. Called w/o a second argument (here: `{}`) it would raise an error (I think…).

`Hash#fetch` is expecially handy when you want to safely go through a nested hash (although I think Ruby 2.3 has a safe navigation operator for that now).

```ruby

my_hash = { hello: 'World' }

# The "usual" way
my_hash[:hello] # => "World"
my_hash[:unknown_key] # => nil

# With fetch
my_hash.fetch(:hello) # "World"
my_hash.fetch(:unknown_key) # boom! 💣

# Return a custom value when key is not found, here: an empty hash
my_hash.fetch(:unknown_key, {}) # => {}

# Use this to chain hashes:
{ nested: { hello: { world: {}, }  } }.fetch(:nested, {}).fetch(:world, {}).fetch(:nonexistent, {}) # => {} 

# Note that these two are the same:
my_hash[:unknown_key] # => nil
my_hash.fetch(:unknown_key, nil) # => nil
```

Errm, does my example make any sense at all?